src/Google.Api.Gax/Api/Gax/ClientHelper.cs

```diff
@@ -0,0 +1,121 @@
+        }
+
+        /// <summary>
+        /// Builds a suitable <see cref="CallOptions"/> taking service settings into account,
+        /// along with a call-specific cancellation token and override mechanism.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token to use for this call.</param>
+        /// <param name="callOptionsOverride">Optional options override delegate.</param>
+        /// <returns></returns>
+        public CallOptions BuildCallOptions(
+            CancellationToken cancellationToken,
+            Func<CallOptions, CallOptions> callOptionsOverride)
+        {
+            var callOptions = new CallOptions(
+                deadline: _deadlineCalculator(),
+                cancellationToken: cancellationToken);
+            return callOptionsOverride?.Invoke(callOptions) ?? callOptions;
+        }
+
+        #region Factory methods for client construction
+        private static async Task<ChannelCredentials> GetScopedApplicationDefaultChannelCredentials(IEnumerable<string> scopes)
+        {
+            var credentials = await GoogleCredential.GetApplicationDefaultAsync().ConfigureAwait(false);
+            if (credentials.IsCreateScopedRequired)
+            {
+                credentials = credentials.CreateScoped(scopes);
+            }
+            return ChannelCredentials.Create(new SslCredentials(), credentials.ToCallCredentials());
+        }
+
+        /// <summary>
+        /// Creates a client using application default credentials, taking care of generic cloning and
+        /// credential scoping.
+        /// </summary>
+        /// <typeparam name="TClient">Type of the client to construct.</typeparam>
+        /// <typeparam name="TSettings">Service-specific settings type.</typeparam>
+        /// <param name="settings">The service settings. It is expected that these have already been cloned.</param>
+        /// <param name="serviceEndpointSettings">The endpoint settings. This is cloned if not null.</param>
+        /// <param name="suppliedCredentialScopes">The credential scopes supplied by the client, if any. These are cloned if not null.</param>
+        /// <param name="defaultCredentialScopes">The credential scopes to apply if none are supplied. These are not expected to change.</param>
+        /// <param name="clientFactory">The factory delegate which creates a client given the relevant credentials, service settings and endpoint.</param>
+        /// <returns>A task which, when completed, will have a result of the newly constructed client.</returns>
+        public static async Task<TClient> CreateFromDefaultCredentialsAsync<TClient, TSettings>(
+            TSettings settings,
+            ServiceEndpointSettings serviceEndpointSettings,
+            IEnumerable<string> suppliedCredentialScopes,
+            IEnumerable<string> defaultCredentialScopes,
+            Func<ChannelCredentials, TSettings, ServiceEndpointSettings, TClient> clientFactory)
```

Comment:
Most of these parameters could be passed into the constructor - settings, serviceEndpointSettings, defaultCredentialScopes, clientFactory
The class would need making generic in TClient and TSettings, which seems fine to me.